cli/unstable_progress_bar.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+/**
+ * The options that are provided to a {@link createProgressBar} or
+ * {@link ProgressBarStream}.
+ */
+export interface ProgressBarOptions {
+  /**
+   * The offset size of the input if progress is resuming part way through.
+   * @default {0}
+   */
+  currentSize?: number;
+  /**
+   * The total size expected to receive.
+   */
+  totalSize: number;
+  /**
+   * The length that the progress bar should be, in characters.
+   * @default {50}
+   */
+  barLength?: number;
+  /**
+   * The character to fill the progress bar up with as it makes progress.
+   * @default {'#'}
+   */
+  fullChar?: string;
+  /**
+   * The character the progress bar starts out with.
+   * @default {'-'}
+   */
+  emptyChar?: string;
+  /**
+   * Whether the progress bar should be removed after completion.
+   * @default {false}
+   */
+  clearBar?: boolean;
```

Comment:
Maybe let's call it just `clear` as in https://github.com/denoland/std/blob/3b75ee7c1925388e01dd69540d51d9387e26dcb0/cli/prompt_secret.ts#L24 ?